### ISSUE TITLE
"-nest" breaks the first signature

### ISSUE THREAD

**Issue by westyles** [mtrojnar/osslsigncode#333]
Hello.
An bug was found during the tests 2.8-dev (from dec 18 2023):

1. The second signature with "-nest" breaks the first signature if the time is different.
Checked several times on different groups of certificates.
https://i.imgur.com/T0D3fGH.png

For example:

```
osslsigncode sign -in file.sys   -out file_2.sys       -spc a.crt -key a.key -nolegacy -h sha1 -time 1420059600 -verbose 

osslsigncode add  -in file_2.sys -out file_3.sys       -h sha1 -TSA-certs TSA1.crt -TSA-key TSA.key -TSA-time 1420059600 -verbose

osslsigncode sign -in file_3.sys -out file_4.sys -nest -spc b.crt -key b.key -pass 0000 -nolegacy -h sha256 -time 1420059601 -TSA-certs TSA2.crt -TSA-key TSA.key -TSA-time 1420059601 -verbose 
```

Additional requests:

2. No way to sign a built-in TSA timestamp on the second signature with a separate operation.
For example, add the "-nest" parameter for "add"

3. Without rewriting for "-out", it became more difficult to perform tests.

**Comment by olszomal** [comment]
@westyles Thank you for reporting this bug! Your detailed information helped me identify and resolve the issue.

`sign -nest` allows to sign file and add the new nested signature instead of replacing the first one, but may be multiplied.
How to identify the correct nested signature to `add -nest`?
```
osslsigncode sign -in unsigned.exe -out signed.exe -certs (...)
osslsigncode sign -in signed.exe -out nested1.exe -nest -certs (...)
osslsigncode sign -in nested1.exe -out nested2.exe -nest -certs (...)
(...)
```
`osslsigncode` opens the input file for reading, and the output file for writing.
It would overwrite its own input if the same file is was used for input and output.

**Comment by westyles** [comment]
> How to identify the correct nested signature to `add -nest`?

I have little understanding of C code and don't know the program's ability to parse signatures.
I think you don't need to think for the user about what signatures are already there, but do the simple thing.
If the program understands the signature itself and its timestamp, then consider them as pairs.
For example:
1 odd: signature, 2 even: timestamp
3 odd: signature, 4 even: timestamp
....

And when "add" + "-nest" is used, set the timestamp to the last odd signature,
replacing the existing timestamp, if there is one and this is possible, not necessarily. And don't sort them!
Basically do the same as "sign" + "-nest".

p.s. Checked "sign -nest", now the first signature doesn't break!
On previous versions of your program, when specifying -out of an existing file, it was deleted just before it started or overwritten.
It was convenient and logical!
I don't know why the stop was made, it doesn't make much sense.
And there is no need to save an existing file for -out.

**Comment by mtrojnar** [comment]
>  I think you don't need to think for the user about what signatures are already there, but do the simple thing.

Please try defining your requirements in a more actionable way:

1. Describe the scenario that doesn't work for you.
2. Describe your preferred command-line syntax to achieve your goal.

Open a new issue for each feature request.
